### ISSUE TITLE
Naming should be consistent for Username

### ISSUE THREAD

**Issue by mgeerling** [realm/realm-studio#1076]
Within the `Users` pane, it seems that naming is not consistent. 

Username seems to be referred to as `Provider ID`, `ID`, and `Username` in different places

Provider:ID 
![image](https://user-images.githubusercontent.com/4967433/50868185-3dbd0a80-1363-11e9-9ef2-312aec86c9a0.png)

ID: 
![image](https://user-images.githubusercontent.com/4967433/50868192-46154580-1363-11e9-8d45-0eadccd69206.png)

Username: 
![image](https://user-images.githubusercontent.com/4967433/50868216-52010780-1363-11e9-87f6-b7a757608b93.png)

**Comment by kraenhansen** [comment]
I agree the naming is a bit confusing, but it's mimicking ROS, if we want to change this we need to find a solution that doesn't make it harder to reason about the system.

In the table of users the "Provider ID(S)" will is actually comma-separated list of ids that the user has in each of the accounts they have associated. On average each user only has a single account associated, but the model allows for a list of accounts. Each account object has a `provider` (the authentication providers name, such as "password", "jwt", etc.) and a `providerId` which is the users id in the context of the particular provider (could be username, email, facebook id, etc.).

The "user id" is ROS internal and unique ID - the provider ids are only unique in the context of a specific provider. I.e. two users could have the same provider id with two different authentication providers.

To simplify the UI I decided to shorten "Provider ID" and write "ID" in the table of accounts, this could be changed to align with the header in the table of users.

When creating a new user, we assume this should be created with the "password" authentication provider and therefore asks for a username (which is the semantic interpretation of the providerId in the context of the password authentication provider).

All of the above to say, there are good reasons for the names above, I would love suggestions on alternatives for these that also doesn't "hide" the internals of ROS - after all Studio is a tool used to administer ROS and therefore also serves as an opportunity for users to understand how the authentication model works.

**Comment by kraenhansen** [comment]
Let's rename the "ID" in the sidebar table to "Provider ID" to align the two and add a tooltip on the tables "Provider ID(s)" column that explains that a user can have multiple accounts and therefore multiple "Provider IDs".

**Comment by roberhofer** [comment]
I have no strong opinion on this. So please go ahead with your suggestion.
